src/Utils.php

```diff
@@ -692,8 +692,9 @@ public static function is_eligible_auto_letterbox( $order ) {
 			$products = $order->get_cart();
 		}
 
-		$is_eligible = self::check_products_for_letterbox( $products );
-
+		if( isset( $products ) ) {
+			$is_eligible = self::check_products_for_letterbox( $products );
+		}
```

Comment:
Please extract cart logic from this function, you can create a new function like :

```php
	public static function is_cart_eligible_auto_letterbox( $cart ) {
		if ( ! in_array( WC()->customer->get_shipping_country(), Utils::get_available_country_for_letterbox(), true ) ) {
			return false;
		}

		return self::check_products_for_letterbox( $cart->get_cart() );
	}
```

and `is_order_eligible_auto_letterbox` function will be like :
```php
	public static function is_order_eligible_auto_letterbox( $order ) {
		if ( wc_get_base_location()['country'] == 'BE' ) {
			return false;
		}

		// Check if order id provided.
		if ( is_int( $order ) ) {
			$order = wc_get_order( $order );
		}

		if ( ! is_a( $order, 'WC_Order' ) ) {
			return false;
		}

		if ( $order->meta_exists( '_postnl_letterbox' ) ) {
			return (bool) $order->get_meta( '_postnl_letterbox', true );
		}

		if ( ! in_array( $order->get_shipping_country(), Utils::get_available_country_for_letterbox(), true ) ) {
			$order->update_meta_data( '_postnl_letterbox', false );
			$order->save_meta_data();

			return false;
		}

		$products    = $order->get_items();
		$is_eligible = self::check_products_for_letterbox( $products );

		$order->update_meta_data( '_postnl_letterbox', $is_eligible );
		$order->save_meta_data();

		return $is_eligible;
	}
```

Reply:
@Abdalsalaam updated